### PR TITLE
Variable Substitution Issue fix

### PR DESCRIPTION
**Pull request by vincentdass**

### CODE REVIEW

File: Tasks/AzureRmWebAppDeployment/Tests/L0.ts [hunk 1]
```diff
@@ -431,7 +431,8 @@ describe('AzureRmWebAppDeployment Suite', function() {
         assert(tr.stdout.search('JSON - simple string change validated') > 0,'JSON -simple string change validation error' );
         assert(tr.stdout.search('JSON - system variable elimination validated') > 0, 'JSON -system variable elimination validation error');
         assert(tr.stdout.search('JSON - special variables validated') > 0, 'JSON - special variables validation error');
-        assert(tr.stdout.search('JSON - varaibles with dot character validated') > 0, 'JSON varaibles with dot character validated');
+        assert(tr.stdout.search('JSON - varaibles with dot character validated') > 0, 'JSON variables with dot character validation error');
```
Comment (arjgupta) [inline Tasks/AzureRmWebAppDeployment/Tests/L0.ts]: This assert should be a part of the corresponding test in IISWebAppDeploymentOnMachineGroup Task as well.

File: Tasks/Common/webdeployment-common/Tests/L1JsonVarSub.ts [hunk 2]
```diff
@@ -1,50 +1,62 @@
...
+}
+if(jsonObject['user.profile']['name.first'] === 'firstName') {
+    console.log('JSON - varaibles with dot character validated');
+}
+if(jsonObject['User.Profile'] === 'do_not_replace') {
+    console.log('JSON - case sensitive variables validated');
+}
+if(jsonObject['constructor.name'] === 'newConstructorName' && 
+    jsonObject['constructor']['name'] === 'newConstructorName' && jsonObject['constructor']['valueOf'] === 'constructorNewValue') {
+        console.log('JSON - substitute inbuilt JSON attributes vaidated');
```
Comment (arjgupta) [inline Tasks/Common/webdeployment-common/Tests/L1JsonVarSub.ts]: minor typo

File: Tasks/AzureRmWebAppDeployment/Tests/L0.ts (hunk 1 above)
Comment (vincentdass) [inline Tasks/AzureRmWebAppDeployment/Tests/L0.ts]: done. thanks

File: Tasks/Common/webdeployment-common/Tests/L1JsonVarSub.ts (hunk 2 above)
Comment (vincentdass) [inline Tasks/Common/webdeployment-common/Tests/L1JsonVarSub.ts]: resolved.

**Pull request merged**